### PR TITLE
Allow dataloader to split data without preprocessing

### LINKED ISSUE #106: getDataset() error
### **Describe the bug:**
Screenshot from 2021-05-21 01-08-09

### **To Reproduce:**
Screenshot from 2021-05-21 01-05-08

### **Expected behavior:**
To create a directory of processed data having training and testing directories.

### **Directory Structure used:**
Screenshot from 2021-05-21 01-04-36

### **Additional context:**

<hr/>

**Note:**
* If you want to work on an issue, you should check if it has already been assigned to anyone.

### PR DESCRIPTION
**Pull request by rubyruins**
# Description:

Originally, dataloader did not split the data into train and test unless augmentation / filters were applied which led to 0 images in the processedData folder. This fix copies original images to the processedData folder if any preprocessed images are not available, thus allowing us to split into train and test sets.

Fixes #106 

## Type of change:

- [x] Bug fix (non-breaking change which fixes an issue)

 
# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.

# Screenshots / Video:

Have verified that old functionality with augmentations and filters works. Have also verified that running just the following 2 lines also works without any preprocessing and the original images get copied to processedData instead:

```
flow = Flow( 'dataset_dir' , 'data_dir' )

(train_x, train_y, val_x, val_y) = flow.getDataset( img_dimensions, test_val_split )
```

### CODE REVIEW

File: flow2ml/Data_Loader.py [hunk 1]
```diff
@@ -80,6 +80,25 @@ def create_dataset(self,classPath):
 
               ind = self.classes.index(folderName)
               self.img_label[image_in_folder] = np.squeeze(np.eye(len(self.classes))[ind])
+    
+    if not any([i != -1 for i in [image.find("Images") for image in os.listdir(classPath)]]):
+
+      for image in list(os.listdir(classPath)):
```
Comment (chebroluharika) [inline flow2ml/Data_Loader.py:86]: Please create a helper function for this copying part. And call that for necessary conditions. Because there is a lot of duplicate code.
Also add comments for the purpose of writing "if" condition.

**Comment by rubyruins** [comment]
@chebroluharika could you check now?

File: flow2ml/Data_Loader.py [hunk 2]
```diff
@@ -64,22 +86,23 @@ def create_dataset(self,classPath):
...
 
-              ind = self.classes.index(folderName)
-              self.img_label[image_in_folder] = np.squeeze(np.eye(len(self.classes))[ind])
+      for image in list(os.listdir(classPath)):
+
+        if not(image.find("Images") != -1):
+        # path is an image since folder could not be found
+        
+          original_image = os.path.join(classPath,image)
+          self.process_images(original_image, final_data_folder, image, classPath)
```
Comment (chebroluharika) [inline flow2ml/Data_Loader.py:105]: How do you get image label without returning anything from process_images()

File: flow2ml/Data_Loader.py [hunk 3]
```diff
@@ -64,22 +86,23 @@ def create_dataset(self,classPath):
...
-              shutil.copy(filtered_image,final_data_folder)
-              
-              # Handle path seperators for Linux and Windows
-              if '/' in classPath:
-                folderName = list(classPath.split("/"))[2]
-              else:
-                folderName = os.path.split(classPath)[1]              
+              self.process_images(filtered_image, final_data_folder, image_in_folder, classPath)
+    
+    if not any([i != -1 for i in [image.find("Images") for image in os.listdir(classPath)]]):
```
Comment (chebroluharika) [inline flow2ml/Data_Loader.py:97]: Add comment here for the inclusion of “if” statement

**Comment by yvkrishna** [comment]
Just a suggession,

In this method, the storage gets decreased as number of images will increase.

**Comment by yvkrishna** [comment]
So can you try to implement in this way !

Loop through each class label folder and put the folder name as a key in dictionary and corresponding image names in that particular folder as a value.

Later while loading the images, based on the key value (i.e class name folder) go to that particular folder and read that image and the next steps are as usual.

File: flow2ml/Data_Loader.py (hunk 2 above)
Comment (rubyruins) [reply flow2ml/Data_Loader.py:105]: self.img_label is a class variable. So it just gets updated for each image everytime the function is called. 

This is what it looks like on printing. All image names and labels are correct:
image

Comment (chebroluharika) [reply flow2ml/Data_Loader.py:105]: Right

**Comment by chebroluharika** [comment]
@rubyruins please take @yvkrishna suggestion and implement it in another PR as this is not related to this bug fix.

**Review by chebroluharika — APPROVED** [review]
